web3.js/src/transaction.ts

```diff
@@ -261,7 +261,7 @@ export class Transaction {
     accountMetas.sort(function (x, y) {
       const checkSigner = x.isSigner === y.isSigner ? 0 : x.isSigner ? -1 : 1;
       const checkWritable =
-        x.isWritable === y.isWritable ? 0 : x.isWritable ? -1 : 1;
+        x.isWritable === y.isWritable ? x.pubkey.toBuffer().compare(y.pubkey.toBuffer()) : x.isWritable ? -1 : 1;
```

Comment:
We also need this ordering for `checkSigner` right?

Reply:
I thought so, but it causes a whole bunch of tests to fail. I'm not sure if those tests were just hardcoded with an ordering in mind, or there's something more sinister at hand.

Reply:
I don't think so. Just spent a bit trying it out. 

The `checkSigner || checkWritable` means that it falls back to the writable ordering. So writable signers go earlier. If both signers aren't writable, then it uses the fallback buffer sorting.

Reply:
Ah yeah, that's right. Thanks!